src/ray/raylet/agent_manager.cc

```diff
@@ -122,6 +122,9 @@ void AgentManager::StartAgent() {
           << "Agent failed to be restarted "
           << RayConfig::instance().agent_max_restart_count()
           << " times. Agent won't be restarted.";
+      if (RayConfig::instance().raylet_shares_fate_with_agent()) {
+        RAY_CHECK(false) << "Raylet shares fate with agent.";
```

Comment:
Can we actually just log with RAY_LOG(ERROR) and use QuickExit() here instead? Ray_CHECK will generate the bad output (that includes cpp stacktrace) to users

Reply:
Done!